include/boost/beast/core/detail/impl/any_buffers.ipp

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright (c) 2016-2021 Vinnie Falco (vinnie dot falco at gmail dot com)
+// Copyright (c) 2021 Richard Hodges (hodges.r@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#include <boost/beast/core/detail/any_buffers.hpp>
+
+namespace boost {
+namespace beast {
+namespace detail {
+
+template<bool IsConst>
+std::size_t
+any_buffers<IsConst>::impl_base::addref() noexcept
+{
+    return ++refs;
+}
+
+template<bool IsConst>
+std::size_t
+any_buffers<IsConst>::impl_base::release() noexcept
+{
+    auto count = --refs;
+    if (count == 0)
+        destroy();
+    return count;
+}
+
+// --------------------------------------------------------------
+
+
+template<bool IsConst>
+any_buffers<IsConst>::any_buffers(any_buffers const& r)
+: impl_(copy(r.impl_))
```

Comment:
if `copy` is only called from here, I would just inline the function into the constructor